src/autoscaler/integration/integration_suite_test.go

```diff
@@ -252,9 +252,33 @@ func stopMetricsServer() {
 	ginkgomon_v2.Kill(processMap[MetricsServerHTTP], 5*time.Second)
 }
 
-func getRandomId() string {
+func getRandomIdRef(ref string) string {
+	report := CurrentSpecReport()
+	// 0123456789012345678901234567890123456789
+	// operator_others:189,11,instance:a5f63cbf7c204c417941d91d21cb3bd0
+	// |15|3|2|10|x| +4 == 40
+
+	id := fmt.Sprintf("%s:%d,%s,%d:%s", testFileFragment(report.FileName()), report.LineNumber(), ref, GinkgoParallelProcess(), randomBits())
+	if len(id) > 40 {
+		id = id[:40]
+	}
+	return id
```

Comment:
Cool idea!

However, with a long enough `ref` it is possible to push out the random part out of the result.
Do we want to to clamp to the constituents down to maximum lengths, so that there's always space for `randomBits`?